### ISSUE TITLE
Flow of the user login

### ISSUE THREAD

**Issue by BernhardSmuts** [aws-amplify/amplify-js#6018]
Authentication

I find it extremely difficult to add a custom user auth flow "out the box" without heavy config.

Current:
-User enters details on auto created UI (Username, Email, password, phone number)
-User gets email
-User needs to enter email code into other waiting page (two pages open, what if user closed the first page)
-User then needs to re-enter login details (on the first page)
-Still has email page open

Ideal:
-User enters only details I need (email, password, confirm password)
-User gets email with verification link
-User clicks on link, is auto signed in and starts using app...

Why is this not the way it's done when this is how 90% of sites do it?

**Comment by elorzafe** [comment]
@BernhardSmuts 

Today we released `autoSignIn` feature on Amplify JS. We will be updating the docs tomorrow, but the feature is available now on version `aws-amplify@4.3.29`

If you are using code verification for new users you only have to do.

```javascript
 Auth.signUp({
      username: 'xxxxxx',
      password: '*********,
      attributes: {
        email: 'xxxxxxxxxx'
      },
      autoSignIn: {
        enabled: true
      }
    })
````

The library will login in the user automatically after the user is confirmed on the App when doing `Auth.confirmSignUp`


If you are using link verification the library also handle that case but you need to configure the library is using `link` verification on `Amplify.configure` like this

```javascript
Amplify.configure({
    Auth: {
       // OPTIONAL
       signUpVerificationMethod: 'link'

        // REQUIRED only for Federated Authentication - Amazon Cognito Identity Pool ID
        identityPoolId: 'XX-XXXX-X:XXXXXXXX-XXXX-1234-abcd-1234567890ab',
        
        // REQUIRED - Amazon Cognito Region
        region: 'XX-XXXX-X',

        // OPTIONAL - Amazon Cognito Federated Identity Pool Region 
        // Required only if it's different from Amazon Cognito Region
        identityPoolRegion: 'XX-XXXX-X',

        // OPTIONAL - Amazon Cognito User Pool ID
        userPoolId: 'XX-XXXX-X_abcd1234',

        // OPTIONAL - Amazon Cognito Web Client ID (26-char alphanumeric string)
        userPoolWebClientId: 'a1b2c3d4e5f6g7h8i9j0k1l2m3',

        // OPTIONAL - Enforce user authentication prior to accessing AWS resources or not
        mandatorySignIn: false,

    }
});
```

Also if the user is autoConfirmed the library will sign in the user as well automatically.

Amplify will emit the event that the user is signed in, or in case you have to use MFA you can use it as well.

```javascript
Hub.listen('auth', ({ payload }) => {
     const { event } = payload;
     if (event === 'autoSignIn') {
          const user = payload.data;
     }
    // ...
})
```

**Comment by abdallahshaban557** [comment]
We also have documentation available here now! https://docs.amplify.aws/lib/auth/emailpassword/q/platform/js/#auto-sign-in-after-sign-up
